airflow/providers/amazon/aws/waiters/databrew.json

```
@@ -0,0 +1,54 @@
+{
+    "version": 2,
+    "waiters": {
+        "job_complete": {
+            "operation": "DescribeJobRun",
+            "delay": 30,
+            "maxAttempts": 60,
+            "acceptors": [
+                {
+                    "matcher": "path",
+                    "argument": "State",
+                    "expected": "STOPPED",
+                    "state": "success"
+                },
+                {
+                    "matcher": "path",
+                    "argument": "State",
+                    "expected": "SUCCEEDED",
+                    "state": "success"
+                },
+                {
+                    "matcher": "path",
+                    "argument": "State",
+                    "expected": "FAILED",
+                    "state": "success"
+                },
+                {
+                    "matcher": "path",
+                    "argument": "State",
+                    "expected": "TIMEOUT",
+                    "state": "success"
+                },
+                {
+                    "matcher": "path",
+                    "argument": "State",
+                    "expected": "Starting",
+                    "state": "retry"
+                },
+                {
+                    "matcher": "path",
+                    "argument": "State",
+                    "expected": "STOPPING",
+                    "state": "retry"
+                },
+                {
+                    "matcher": "path",
+                    "argument": "State",
+                    "expected": "RUNNING",
+                    "state": "retry"
+                }
```

```suggestion
                }
```

There are only two available options of state in botocore waiter `success` or `failure`, if you need to keep waiting (e.g. intermediate states) then you do not need add anything for the matcher

https://github.com/boto/botocore/blob/f0ff061aebd9ef15a66f87b675789275e373da43/botocore/waiter.py#L366-L371

---

Hmmm…. I was check internal botocore waiters and found that some waiters use `retry` in case if matcher is an error https://github.com/boto/botocore/blob/master/botocore/data/ec2/2016-11-15/waiters-2.json

---

Interesting. I'll make the change. Not sure if retrying errors in this case makes sense (except maybe TIMEOUT), I'm assuming EC2 does it to allow for resources to be created.

---

Unfortunetly there is no documentation how to create custom waiters provided by boto Team.
So it always grey area `¯\_(ツ)_/¯`